packages/compass-components/src/components/error-warning-summary.tsx

```diff
@@ -9,7 +8,7 @@ import { Banner, Button } from './leafygreen';
 type ErrorOrWarning = { message: string };
 
 const bannerStyle = css({
-  marginTop: spacing[1],
+  width: '100%',
```

Comment:
I like this change, can we update the connection-form to have a margin above the banner then? 
<img width="632" alt="Screen Shot 2022-05-06 at 4 05 23 PM" src="https://user-images.githubusercontent.com/1791149/167209467-8b4bb1e4-4348-4a62-80ae-3f24e22eeb26.png">

Reply:
Yeah, so if we want these components to be reusable, we should be careful about putting positioning styles on them, this makes them very hard to reuse and this stuff will almost always be different. This one is also controlling its own render which makes it tricky to pass any additional margin to it. Would you prefer I add a `className` to it that we can pass down or remove this implicit show / hide logic and move it to the place where its used and wrap it in something with appropriate padding? I'd probably prefer the latter as this decouples the layouting part from the component logic, but don't have that strong of an opinion. Maybe you have other ideas how we can make this component reusable and bring back the padding in the connection form

Reply:
I do like that the component abstracts away the show/hide logic so not every consumer has to use it. At the same time `className` has implications that the consumers is then knowing a bit too much about the component. I agree with you that removing the show/hide logic and having it in the places where the summary component used is a nicer solution, and gives less coupling, although it will be a bit more coding in places. I would prefer that one, also not a strong preference. Can't think of a nicer way really, one other alternative would be a `withTopPadding` as a boolean prop. I think we've been avoiding that method of component usage though, and it could lead down the path of having tons of slight variations on a component.

Reply:
Agree, I'd try to avoid doing things like `withPadding` for same reasons. Okay, I'll go them with a way that we both seem to lean to, thanks for bouncing ideas around!